src/manifest.c

```diff
@@ -1425,36 +1432,55 @@ int enforce_compliant_manifest(struct file **a, struct file **b, int searchsize,
 
 /* The function splits a list of manifests into two lists:
  * - full_manifests: this list will contain all the full manifests that were in the manifests list
- * - iterative_manifests: this list will contain all the valid iterative manifests from the manifests
+ * - consolidated_manifests: this list will contain all the valid iterative manifests from the manifests
  *   list, plus the full manifest of those bundles that didn't have a valid iterative manifest
  *   A valid iterative manifest is one that can be used during an update, meaning that the iterative
  *   manifest must match the "from version" provided, and that it belongs to an already installed bundle*/
-void filter_iterative_manifests(struct list *manifests, int version, struct list **full_manifests, struct list **iterative_manifests)
+void consolidate_manifests(struct list *manifests, int version, struct list **consolidated_manifests)
 {
-	/* Split the manifests into "iterative manifests" and "full manifests" */
+	struct list *full_manifests = NULL;
 	struct list *list;
 	struct file *file;
 
+	/* Split the manifests into "iterative manifests" and "full manifests" */
 	for (list = manifests; list; list = list->next) {
 		file = list->data;
 		/* Keep only iterative manifests that are from the specified version and that are
 		 * already installed */
 		if (file->is_iterative == 1 && file->from_version == version && is_tracked_bundle(file->bundlename)) {
-			*iterative_manifests = list_prepend_data(*iterative_manifests, file);
+			*consolidated_manifests = list_prepend_data(*consolidated_manifests, file);
 		} else if (file->is_iterative != 1) {
-			*full_manifests = list_prepend_data(*full_manifests, file);
+			full_manifests = list_prepend_data(full_manifests, file);
 		}
 	}
 
 	/* Add the full manifest of those bundles that do not have a valid iterative manifest
-	* into the list of iterative_manifests */
-	for (list = *full_manifests; list; list = list->next) {
+	* into the list of consolidated_manifests */
+	for (list = full_manifests; list; list = list->next) {
 		file = list->data;
-		if (!search_bundle_in_manifests(*iterative_manifests, file->bundlename)) {
-			*iterative_manifests = list_prepend_data(*iterative_manifests, file);
+		if (!search_bundle_in_manifests(*consolidated_manifests, file->bundlename)) {
+			*consolidated_manifests = list_prepend_data(*consolidated_manifests, file);
```

Comment:
I think this for loop could be consolidated into the one above.  When adding an item to the full_manifests list, we could check whether it should be added to the consolidated_manifests list too.

Reply:
You need to have all the valid iterative manifests first in the list of consolidated manifests or you would not know what full manifests to add to it. After all you only need to add those full manifests of bundles that didn't have an iterative one.

Reply:
You're right, I see now.